lib/promote_release.js

```diff
@@ -0,0 +1,437 @@
+    await forceRunAsync('git', ['checkout', 'HEAD', '--', 'src/node_version.h'],
+      { ignoreFailure: false });
+
+    // There will be remaining cherry-pick conflicts the Releaser will
+    // need to resolve, so confirm they've been resolved before
+    // proceeding with next steps.
+    cli.separator();
+    cli.info('Resovle the conflicts and commit the result');
+    cli.separator();
+    const didResolveConflicts = await cli.prompt(
+      'Finished resolving cherry-pick conflicts?', { defaultAnswer: true });
+    if (!didResolveConflicts) {
+      cli.warn(`Aborting release promotion for version ${version}`);
+      throw new Error('Aborted');
+    }
+
+    // Validate release commit on the default branch
+    const releaseCommitOnDefaultBranch =
+      await forceRunAsync('git', ['show', 'HEAD', '--name-only', '--pretty=format:%s'],
+        { captureStdout: true, ignoreFailure: false });
+    const [commitTitle, ...modifiedFiles] = releaseCommitOnDefaultBranch.trim().split('\n');
+    await this.validateReleaseCommit(commitTitle);
+    if (modifiedFiles.some(file => !file.endsWith('.md'))) {
+      cli.warn('Some modified files are not markdown, that\'s unusual.');
+      cli.info(`The list of modified files: ${modifiedFiles.map(f => `- ${f}`).join('\n')}`);
+      if (!await cli.prompt('Do you want to proceed anyway?', { defaultAnswer: false })) {
+        throw new Error('Aborted');
+      }
+    }
+
+    // Push to the remote default branch
+    const shouldPushDefaultBranch = await cli.prompt(`Push to ${this.branch}?`,
+      { defaultAnswer: true });
+    if (!shouldPushDefaultBranch) {
+      cli.warn(`Aborting release promotion for version ${version}`);
+      throw new Error('Aborted');
+    }
+    if (this.dryRun) {
+      cli.info(`Skipping pushing to ${this.branch} in dry-run mode`);
+    } else {
+      await forceRunAsync('git', ['push', this.upstream, this.branch], { ignoreFailure: false });
+    }
+
+    // Push release tag.
+    const shouldPushTag = await cli.prompt('Push release tag?',
+      { defaultAnswer: true });
+    if (!shouldPushTag) {
+      cli.warn(`Aborting release promotion for version ${version}`);
+      throw new Error('Aborted');
+    }
+    await this.pushReleaseTag();
+
+    // Promote and sign the release builds.
+    const shouldPromote = await cli.prompt('Promote and sign release builds?',
+      { defaultAnswer: true });
+    if (!shouldPromote) {
+      cli.warn(`Aborting release promotion for version ${version}`);
+      throw new Error('Aborted');
+    }
+
+    // TODO: move this to .ncurc
+    const defaultKeyPath = '~/.ssh/node_id_rsa';
+    const keyPath = await cli.prompt(
+      `Please enter the path to your ssh key (Default ${defaultKeyPath}): `,
+      { questionType: 'input', defaultAnswer: defaultKeyPath });
+    await this.promoteAndSignRelease(keyPath);
+
+    cli.separator();
+    cli.ok(`Release promotion for ${version} complete.\n`);
+    cli.info(
```

Comment:
Maybe add a TODO to automate all these steps too?

Reply:
I think the only step that can be automated is "Create the release on GitHub" (although simply printing out which `gh` command to use might be preferable, it might be a bad idea to have NCU token with permission to create releases), I'll see if I can add it – the "Create the blog post for nodejs.org" step might be automatable as well, but a cross-repo script call would feel quite brittle.

Reply:
https://github.com/nodejs/node-core-utils/pull/835/commits/66fdab573af0456ae0f487597dee0f9a0d9f0aa2